target/linux/layerscape/image/mkits-multiple-config.sh

```diff
@@ -0,0 +1,594 @@
+#	2: image index
+#	3: loadaddr of image
+#	4: entrypoint of image
+#	5: compression
+#	6: hash algorithm
+#	7: part of the configuration
+#	8: Human friend name for the image
+#	9: key file name
+#	10: signature
+# 	11: conf friendly name
+
+# confX array index information
+#	0: conf number
+#	1: kernel conf
+#	2: fdt conf
+#	3: rootfs conf
+#	4: kernel key file
+#	5: fdt key file
+#	6: rootfs key file
+#	7: kernel sign_algorithm
+#	8: fdt sign_algorithm
+#	9: rootfs sign_algorithm
+#	10: conf friendly name
+
+usage() {
+	echo "Usage: `basename $0` -A arch -v version -o its_file" \
+		"-k kernel -a addr -e entry [-C none] [-h sha1] [-c conf]"
+	echo -e "Example1:\n\tkernel image ker_img1 with no compression +"
+	echo -e "\tsha1 hash + fdt dtb1 with sha1 and crc32 hash for conf 1"
+	echo -e "\t $ `basename $0` -A arm -v 4.4 \ "
+	echo -e "\t      -k ker_img1 -C none -h sha1 -e 0x8000 -a 0x8000 -c 1 \ "
+	echo -e "\t      -d dtb1 -h sha1 -h crc32 -c 1\n"
+	echo "General settings:"
+	echo -e "\t-A ==> set architecture to 'arch'"
+	echo -e "\t-v ==> set kernel version to 'version'"
+	echo -e "\t-o ==> create output file 'its_file' [optional]"
+	echo "Input image type:"
+	echo -e "\t-k ==> kernel image 'kernel'"
+	echo -e "\t-d ==> Device Tree Blob 'dtb'"
+	echo -e "\t-r ==> ramdisk image 'ramdisk"
+	echo "Per image configurations:"
+	echo -e "\t-C ==> set compression type 'comp'"
+	echo -e "\t-c ==> set image config (multiple -c allowed)"
+	echo -e "\t-a ==> set load address to 'addr' (hex)"
+	echo -e "\t-e ==> set entry point to 'entry' (hex)"
+	echo -e "\t-D ==> human friendly 'name' (one word only)"
+	echo -e "\t-h ==> set hash algorithm (multiple -h allowed)"
+	echo -e "\t-s ==> set signature for given config image"
+	echo -e "\t-K ==> set key file for given config image"
+	exit 1
+}
+
+array_check()
+{
+	local a=999
+	local max_a=0
+	local max_i=0
+
+	if echo $1 | grep -q img; then
+		max_a=$MAX_IMG
+		max_i=13
+		let a=$(echo $1 | awk -F "img" '{print $2}')
+	elif echo $1 | grep -q conf; then
+		max_a=$MAX_CONF
+		max_i=10
+		let a=$(echo $1 | awk -F "conf" '{print $2}')
+	fi
+	if [ ${a} -lt 0 -o ${a} -gt ${max_a} -o \
+		${2} -lt 0 -o ${2} -gt ${max_i} ]; then
+		echo "WARNING: Invalid array name, skipping!!!"
+		return 255
+	fi
+}
+
+#
+# $1:	array name
+# $2:	index
+# $3:	value
+# $4:	append operation
+#
+array_put()
+{
+	# check if array is declared
+	array_check $1 $2 || return 0
+	if [ -z "$4" ]; then
+		eval $1[$2]=$3
+	else
+		eval $1[$2]=\"\${$1[$2]} $3\"
+	fi
+}
+
+#
+# $1:	array name
+# $2:	index
+#
+array_get()
+{
+	local val
+	eval val=\${$1[$2]}
+	echo $val
+}
+
+parse_args() {
+	local i=-1 k=-1 d=-1 r=-1
+	while getopts ":A:a:C:c:D:d:e:h:k:K:o:v:r:s:n:" OPTION; do
+		case $OPTION in
+			A ) ARCH=$OPTARG;;
+			a ) array_put img$i 3 $OPTARG;;
+			C ) value_sanity_chk compression $OPTARG;
+				array_put img$i 5 $OPTARG;;
+			c ) array_put img$i 7 $OPTARG append;;
+			D ) array_put img$i 8 $OPTARG;;
+			d ) i=$(($i + 1));
```

Comment:
Don't need semis on any of these lines since you don't have multiple statements on the same line.